crates/test-programs/wasi-tests/src/bin/path_filestat.rs

```diff
@@ -117,8 +117,61 @@ unsafe fn test_path_filestat(dir_fd: wasi::Fd) {
     assert_eq!(stat.mtim, new_mtim, "mtim should not change");
     assert_eq!(stat.atim, old_atim, "atim should not change");
 
+    // Create a symlink
+    wasi::path_symlink("file", dir_fd, "symlink").expect("creating symlink to a file");
+
+    // Check path_filestat_set_times on the symlink itself
+    let mut sym_stat = wasi::path_filestat_get(dir_fd, 0, "file").expect("reading file stats");
+
+    let sym_new_mtim = sym_stat.mtim - 200;
+    wasi::path_filestat_set_times(
+        dir_fd,
+        0,
+        "symlink",
+        // on purpose: the syscall should not touch atim, because
+        // neither of the ATIM flags is set
+        sym_new_mtim,
+        sym_new_mtim,
+        wasi::FSTFLAGS_MTIM,
+    )
+    .expect("path_filestat_set_times should succeed on symlink");
+
+    sym_stat = wasi::path_filestat_get(dir_fd, 0, "symlink")
+        .expect("reading file stats after path_filestat_set_times");
+    assert_eq!(sym_stat.mtim, new_mtim, "mtim should change");
+    assert_eq!(sym_stat.atim, old_atim, "atim should not change");
```

Comment:
I realize this is copied from other tests in this file, but the `atim` field is effectively volatile; it could change depending on factors outside the control of this test, which could cause this check to fail, so unfortunately I think we should omit these checks.

Reply:
Makes sense! I've addressed that in the latest commit.